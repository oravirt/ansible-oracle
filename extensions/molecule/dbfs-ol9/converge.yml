---
- name: Converge
  hosts: all
  gather_facts: true
  any_errors_fatal: true
  vars:
    patch_before_rootsh: true
    apply_patches_db: true

    db_homes_installed:
      - home: db19-si-ee
        state: present
        apply_patches: true
  tasks:

    # install missing cron on rhel9 container
    - name: Install cron on RHEL/OL9
      ansible.builtin.package:
        name: cronie
      when:
        - ansible_os_family == 'RedHat'
        - ansible_distribution_major_version | int == 9

- name: Converge os
  ansible.builtin.import_playbook: opitzconsulting.ansible_oracle.os

- name: Converge swdb
  ansible.builtin.import_playbook: opitzconsulting.ansible_oracle.swdb
  vars:
    # install it directly from remote in this test
    # => Do not copy the installation media into container
    oracle_sw_copy: true
    is_sw_source_local: true

- name: Converge sql_zauberkastern
  ansible.builtin.import_playbook: opitzconsulting.ansible_oracle.sql_zauberkastern

- name: Converge manage_db
  ansible.builtin.import_playbook: opitzconsulting.ansible_oracle.manage_db

- name: Converge manage_ords
  ansible.builtin.import_playbook: opitzconsulting.ansible_oracle.manage_ords
