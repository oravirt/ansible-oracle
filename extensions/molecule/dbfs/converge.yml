---
- name: Converge
  hosts: all
  gather_facts: true
  any_errors_fatal: true
  tasks:

    # install missing cron on rhel9 container
    - name: Install cron on RHEL/OL9
      ansible.builtin.package:
        name: cronie
      when:
        - ansible_os_family == 'RedHat'
        - ansible_distribution_major_version | int == 9

    # install missing packages for openSUSE
    - name: Install packages on openSUSE
      ansible.builtin.package:
        name:
          - cronie
          - git
          - lsof
          - ssh-tools
      when:
        - ansible_os_family == 'Suse'

- name: Converge os
  ansible.builtin.import_playbook: opitzconsulting.ansible_oracle.os

- name: Converge swdb
  ansible.builtin.import_playbook: opitzconsulting.ansible_oracle.swdb
  vars:
    # install it directly from remote in this test
    # => Do not copy the installation media into container
    oracle_sw_copy: false
    is_sw_source_local: true
    oracle_stage_remote: /vagrant

- name: Converge sql_zauberkastern
  ansible.builtin.import_playbook: opitzconsulting.ansible_oracle.sql_zauberkastern

- name: Converge manage_db
  ansible.builtin.import_playbook: opitzconsulting.ansible_oracle.manage_db
