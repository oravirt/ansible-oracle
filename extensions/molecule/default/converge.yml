---
- name: Converge
  hosts: all
  gather_facts: true
  any_errors_fatal: true
  tasks:
    - name: "Import common"
      ansible.builtin.import_role:
        name: opitzconsulting.ansible_oracle.common

    - name: "Import orahost"
      ansible.builtin.import_role:
        name: "opitzconsulting.ansible_oracle.orahost"

    - name: "Import orahost_storage"
      ansible.builtin.import_role:
        name: "opitzconsulting.ansible_oracle.orahost_storage"

    - name: "Import cxoracle"
      ansible.builtin.import_role:
        name: "opitzconsulting.ansible_oracle.cxoracle"

    - name: "Import orahost_logrotate"
      ansible.builtin.import_role:
        name: "opitzconsulting.ansible_oracle.orahost_logrotate"
