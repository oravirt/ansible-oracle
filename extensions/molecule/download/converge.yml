---
- name: Converge
  hosts: all
  gather_facts: true
  any_errors_fatal: true
  vars:
    mos_login: "{{ lookup('ansible.builtin.env', 'MOS_LOGIN', default=Undefined) }}"
    mos_password: "{{ lookup('ansible.builtin.env', 'MOS_PASSWORD', default=Undefined) }}"
  tasks:
    # - debug: var=oracle_sw_patches
    - name: Assert credentials
      ansible.builtin.assert:
        quiet: true
        that:
          - mos_login is defined
          - mos_password is defined
        fail_msg: "Set MOS_LOGIN and MOS_PASSWORD in shell environment and restart converge again."

    - name: "Import orasw_download_patches"
      ansible.builtin.import_role:
        name: opitzconsulting.ansible_oracle.orasw_download_patches
