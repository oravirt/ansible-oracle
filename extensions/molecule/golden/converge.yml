---
- name: Converge os
  ansible.builtin.import_playbook: opitzconsulting.ansible_oracle.os

- name: Converge swdb
  ansible.builtin.import_playbook: opitzconsulting.ansible_oracle.swdb
  vars:
    # install it directly from remote in this test
    # => Do not copy the installation media into container
    oracle_sw_copy: false
    is_sw_source_local: true
    oracle_stage_remote: /vagrant

- name: Converge - DB-Patching
  ansible.builtin.import_playbook: opitzconsulting.ansible_oracle.opatch
  vars:
    # the shared inventory has configured databases.
    # => Datapatch will try to start a database in this playbook
    # => Remove them in this scenario, because we do not need any database
    oracle_databases: []
    oracle_pdbs: []

- name: Converge oraswdb_golden_image
  ansible.builtin.import_playbook: opitzconsulting.ansible_oracle.golden_images
